public/app/features/dashboard-scene/scene/ScopesFiltersScene.tsx

```diff
@@ -9,111 +10,269 @@ import {
+  const { nodes, expandedNodes, scopes } = model.useState();
   const parentState = model.parent!.useState();
   const isViewing = 'isViewing' in parentState ? !!parentState.isViewing : false;
 
-  const options: Array<SelectableValue<string>> = scopes.map(({ metadata: { name }, spec: { title, category } }) => ({
-    label: title,
-    value: name,
-    description: category,
-  }));
+  return (
+    <Toggletip
+      content={
+        <ScopesTreeLevel
+          isExpanded
+          path={[]}
+          nodes={nodes}
+          expandedNodes={expandedNodes}
+          scopes={scopes}
+          model={model}
+        />
+      }
+      footer={'Open advanced scope selector'}
+      closeButton={false}
+    >
+      <Input disabled={isViewing} readOnly value={scopes.map((scope) => scope.spec.title)} />
+    </Toggletip>
+  );
+}
+
+export interface ScopesTreeLevelProps {
+  isExpanded: boolean;
+  path: string[];
+  nodes: Record<string, Node>;
+  expandedNodes: string[];
+  scopes: Scope[];
+  model: ScopesFiltersScene;
+}
+
+export function ScopesTreeLevel({ isExpanded, path, nodes, expandedNodes, scopes, model }: ScopesTreeLevelProps) {
+  const styles = useStyles2(getStyles);
+
+  if (!isExpanded) {
+    return null;
+  }
 
   return (
-    <Select
-      isClearable
-      isLoading={isLoading}
-      disabled={isViewing}
-      options={options}
-      value={value}
-      onChange={(selectableValue) => model.setScope(selectableValue?.value ?? undefined)}
-    />
+    <div role="tree" className={path.length > 0 ? styles.innerLevelContainer : undefined}>
+      {Object.values(nodes).map((node) => {
+        const {
+          item: { nodeId, linkId },
+          isScope,
+          children,
+        } = node;
+        const nodePath = [...path, nodeId];
+        const isExpanded = expandedNodes.includes(nodeId);
+        const isSelected = isScope && !!scopes.find((scope) => scope.metadata.name === linkId);
+
+        return (
+          <div
+            key={nodeId}
+            role="treeitem"
+            aria-selected={isExpanded}
+            tabIndex={0}
+            className={cx(styles.item, isScope && styles.itemScope)}
+            onClick={(evt) => {
+              evt.stopPropagation();
+              model.expandNode(nodePath);
+            }}
+            onKeyDown={(evt) => {
+              evt.stopPropagation();
+              model.expandNode(nodePath);
+            }}
+          >
+            {!isScope ? (
+              <Icon className={styles.icon} name="folder" />
+            ) : (
+              <Checkbox
+                className={styles.checkbox}
+                checked={isSelected}
+                onChange={(evt) => {
+                  evt.stopPropagation();
+
+                  if (linkId) {
+                    model.toggleScope(linkId);
+                  }
+                }}
+              />
+            )}
+
+            <span>{node.item.title}</span>
```

Comment:
UX: I miss the possibility to click on node item, I need to be very precise to click on the checkbox.

Reply:
Yeah, I tried to work around this even for the showcase but it gets a little weird from an accessibility perspective. Once we have the final designs, I'll find a way to overcome this while implementing them.